packages/std/src/math/isqrt.rs

```diff
@@ -59,19 +59,19 @@ mod tests {
     #[test]
     fn isqrt_primitives() {
         // Let's check correctness.
-        assert_eq!(0u8.isqrt(), 0);
-        assert_eq!(1u8.isqrt(), 1);
-        assert_eq!(24u8.isqrt(), 4);
-        assert_eq!(25u8.isqrt(), 5);
-        assert_eq!(26u8.isqrt(), 5);
-        assert_eq!(36u8.isqrt(), 6);
+        assert_eq!(super::Isqrt::isqrt(0u8), 0);
+        assert_eq!(super::Isqrt::isqrt(1u8), 1);
+        assert_eq!(super::Isqrt::isqrt(24u8), 4);
+        assert_eq!(super::Isqrt::isqrt(25u8), 5);
+        assert_eq!(super::Isqrt::isqrt(26u8), 5);
+        assert_eq!(super::Isqrt::isqrt(36u8), 6);
 
         // Let's also check different types.
-        assert_eq!(26u8.isqrt(), 5);
-        assert_eq!(26u16.isqrt(), 5);
-        assert_eq!(26u32.isqrt(), 5);
-        assert_eq!(26u64.isqrt(), 5);
-        assert_eq!(26u128.isqrt(), 5);
+        assert_eq!(super::Isqrt::isqrt(26u8), 5);
+        assert_eq!(super::Isqrt::isqrt(26u16), 5);
+        assert_eq!(super::Isqrt::isqrt(26u32), 5);
+        assert_eq!(super::Isqrt::isqrt(26u64), 5);
+        assert_eq!(super::Isqrt::isqrt(26u128), 5);
```

Comment:
Why are those changed needed? Which rule is that? Does not really look like an improvement.

Reply:
```
error: a method with this name may be added to the standard library in the future
  --> packages/std/src/math/isqrt.rs:62:24
   |
62 |         assert_eq!(0u8.isqrt(), 0);
   |                        ^^^^^
   |
   = warning: once this associated item is added to the standard library, the ambiguity may cause an error or change in behavior!
   = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
   = help: call with fully qualified syntax `math::isqrt::Isqrt::isqrt(...)` to keep using the current method
   = help: add `#![feature(isqrt)]` to the crate attributes to enable `core::num::<impl u8>::isqrt`
   = note: `-D unstable-name-collisions` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unstable_name_collisions)]`
```

Reply:
It's needed because when rust std adds the `isqrt` method, the test will silently (except for maybe an unused import warning) start testing that instead of our trait impl

Reply:
Ah, nice, thanks!